src/field/captcha/third_party_captcha.jsx

```diff
@@ -84,39 +119,88 @@ export class ThirdPartyCaptcha extends React.Component {
     };
   }
 
-  static loadScript(props, element = document.body, callback = noop) {
-    const callbackName = `${providerDomPrefix(props.provider)}Callback_${Math.floor(Math.random() * 1000001)}`;
-    const scriptUrl = scriptForProvider(props.provider, props.hl, callbackName);
-    const script = document.createElement('script');
-
-    window[callbackName] = () => {
-      delete window[callbackName];
-      callback(null, script);
+  injectCaptchaScript(callback = noop) {
+    const { provider, hl, clientSubdomain, sitekey } = this.props;
+    const callbackName = `${providerDomPrefix(provider)}Callback_${Math.floor(Math.random() * 1000001)}`;
+    const scriptUrl = scriptForProvider(provider, hl, callbackName, clientSubdomain, sitekey);
+    const attributes = {
+      async: true,
+      defer: true
     };
-
-    script.src = scriptUrl;
-    script.async = true;
-    script.defer = true;
-    if (props.provider === FRIENDLY_CAPTCHA_PROVIDER) {
-      script.onload = window[callbackName];
+    if (provider === ARKOSE_PROVIDER) {
+      attributes['data-callback'] = callbackName;
+      attributes['onerror'] = function () {
+        if (this.state.retryCount < MAX_RETRY) {
+          removeScript(scriptUrl);
+          loadScript(scriptUrl, attributes);
+          this.setState((prevState) => ({
+            retryCount: prevState.retryCount + 1
+          }));
+          return;
+        }
+        removeScript(scriptUrl);
+        this.changeHandler('BYPASS_CAPTCHA');
+      };
+      window[callbackName] = function (arkose) {
+        callback(arkose);
+      };
+    } else {
+      window[callbackName] = () => {
+        delete window[callbackName];
+        callback();
+      };
+  
+      if (provider === FRIENDLY_CAPTCHA_PROVIDER) {
+        attributes['onload'] = window[callbackName];
+      }
     }
 
-    element.appendChild(script);
+    loadScript(scriptUrl, attributes);
   }
 
   componentWillUnmount() {
-    if (!this.scriptNode) {
+    if (!this.state.scriptUrl) {
       return;
     }
-    document.body.removeChild(this.scriptNode);
+    removeScript(this.state.scriptUrl);
   }
 
   componentDidMount() {
-    ThirdPartyCaptcha.loadScript(this.props, document.body, (err, scriptNode) => {
-      this.scriptNode = scriptNode;
+    this.injectCaptchaScript((arkose) => {
       const provider = getCaptchaProvider(this.props.provider);
-
-      if (this.props.provider === FRIENDLY_CAPTCHA_PROVIDER) {
+      if (this.props.provider === ARKOSE_PROVIDER) {
+        var handleArkose = function (event) {
+          setTimeout(function () {
+            arkose.run();
+          }, TIMEOUT_MS);
+          // TODO: This should prevent the form from submitting
```

Comment:
Can we ensure the TODO is either resolved, or removed if no longer necessary?

Reply:
So the reason I left the TODO is because this is not working as intended yet. `event.preventDefault` won't stop the login request. We want to halt the request and run captcha challenge and then once the challenge is completed, we want to submit the form.

Reply:
I think it's because this is how we submit: https://github.com/auth0/lock/blob/f89158391203d4c6cf99e8621928e1ac2e94c71d/src/ui/box/container.jsx#L224 and the code here is conflicting with that as we have two submit handlers.

I would think you need to find a way to avoid the custom submit here, but wire in the captcha on that level.

Reply:
Not seeing a simple solution with the current approach. This may be quite difficult to do and might require lots of refactoring. One simple solution would be to just trigger the captcha on page load, then we don't have to worry about listening on the form submit.